pkg/cli/gettoken/gettoken.go

```diff
@@ -0,0 +1,269 @@
+			kcmdutil.CheckErr(o.Validate())
+			kcmdutil.CheckErr(o.Run())
+		},
+	}
+
+	cmd.Flags().StringVar(&o.IssuerURL, "issuer-url", o.IssuerURL, "Issuer URL of the external OIDC provider")
+	cmd.Flags().StringVar(&o.ClientID, "client-id", o.ClientID, "Client ID of the user managed by the external OIDC provider")
+	cmd.Flags().StringVar(&o.ClientSecret, "client-secret", o.ClientSecret, "Client Secret of the user managed by the external OIDC provider. Optional.")
+	cmd.Flags().StringSliceVar(&o.ExtraScopes, "extra-scopes", o.ExtraScopes, "Extra scopes for the auth request to the external OIDC provider. Optional.")
+	cmd.Flags().StringSliceVar(&o.CallbackAdresses, "callback-addresses", o.CallbackAdresses, "Callback addresses where external OIDC issuer redirects to after flow is completed. If multiple is passed, command will try to use in an order. Defaults to localhost:8000.")
+	cmd.Flags().DurationVar(&o.AuthenticationTimeout, "authentication-timeout", o.AuthenticationTimeout, "Duration indicates that how long command should wait the authe flow is completed. Defaults to 3 minutes.")
+	cmd.Flags().BoolVar(&o.AutoOpenBrowser, "auto-open-browser", o.AutoOpenBrowser, "Specify browser is automatically opened or not.")
+
+	return cmd
+}
+
+func (o *GetTokenOptions) Complete(f kcmdutil.Factory, cmd *cobra.Command, args []string) error {
+	o.CACertFilename = kcmdutil.GetFlagString(cmd, "certificate-authority")
+	o.InsecureTLS = kcmdutil.GetFlagBool(cmd, "insecure-skip-tls-verify")
+
+	provider := oidc.Provider{
+		IssuerURL:    o.IssuerURL,
+		ClientID:     o.ClientID,
+		ClientSecret: o.ClientSecret,
+		ExtraScopes:  o.ExtraScopes,
+		UsePKCE:      true,
+	}
+
+	authenticator, err := oidc.NewAuthenticator(context.TODO(), provider, "", o.CACertFilename, o.InsecureTLS)
+	if err != nil {
+		return fmt.Errorf("oidc authenticator error %v", err)
+	}
+
+	o.authenticator = authenticator
+	o.tokenCache = &tokencache.Repository{}
+	o.credWriter = credwriter.NewWriter(o.IOStreams)
+
+	if kcd := os.Getenv("KUBECACHEDIR"); kcd != "" {
+		o.tokenCacheDir = filepath.Join(kcd, "oc")
+	} else {
+		o.tokenCacheDir = filepath.Join(homedir.HomeDir(), ".kube", "cache", "oc")
+	}
+
+	return nil
+}
+
+func (o *GetTokenOptions) Validate() error {
+	if o.IssuerURL == "" {
+		return fmt.Errorf("--issuer-url is required")
+	}
+	if o.ClientID == "" {
+		return fmt.Errorf("--client-id is required")
+	}
+
+	return nil
+}
+
+// Run starts the authentication flow with a caching and refreshing capability.
+// If refresh token is found, it tries to use it to get a valid id token from
+// external OIDC issuer. If not, it forces user to log in.
+func (o *GetTokenOptions) Run() error {
+	tokenCacheKey := tokencache.Key{
+		IssuerURL:      o.IssuerURL,
+		ClientID:       o.ClientID,
+		ExtraScopes:    o.ExtraScopes,
+		CACertFilename: o.CACertFilename,
+		SkipTLSVerify:  o.InsecureTLS,
+	}
+
+	tokenSet, err := o.tokenCache.FindByKey(o.tokenCacheDir, tokenCacheKey)
```

Comment:
We can safely ignore the error. Because if there is any error occurred other than missed cache, it will raise error during writing to cache. I updated by ignoring error with a explanation. Let me know what you think.